lib/src/multi_sig/multi_sig_builder.ts

```diff
@@ -0,0 +1,329 @@
+import ow from 'ow';
+import { NetworkEnum } from '../network';
+import { PublicKey } from '../types';
+
+const native = require('../../../native');
+
+/**
+ * builder to facilitate multiSig flow
+ * @class MultiSigBuilder
+ */
+export class MultiSigBuilder {
+    private keyPair: {
+        privateKey?: Buffer;
+        publicKey?: Buffer;
+    };
+
+    private network: NetworkEnum;
+
+    private signerPublicKeys: Array<PublicKey> = [];
+
+    private requiredSignerNumber: number;
+
+    // transferable address start with `cro1`
+    public multiSigAddress?: string;
+
+    // 32 bytes message for session creation. E.g. fund tx hash
+    public message?: string;
+
+    // current session, could be complete or incomplete
+    private session?: string;
+
+    // self nonce commitment
+    public nonceCommitment?: string;
+
+    // local cache for the already added co-signer's commitments
+    private coSignerNonceCommitments?: Array<string> = [];
```

Comment:
Since there is default value of empty array, the optional flag can be removed
```suggestion
    private coSignerNonceCommitments: Array<string> = [];
```